src/main/java/com/hivemq/embedded/EmbeddedExtensionBuilder.java

```diff
@@ -0,0 +1,125 @@
+import com.hivemq.extension.sdk.api.ExtensionMain;
+import com.hivemq.extension.sdk.api.annotations.DoNotImplement;
+import com.hivemq.extension.sdk.api.annotations.NotNull;
+
+import javax.annotation.Nullable;
+
+/**
+ * @author Florian Limpöck
+ */
+@DoNotImplement
+public interface EmbeddedExtensionBuilder {
+
+    /**
+     * @return a new EmbeddedExtensionBuilder.
+     */
+    static @NotNull EmbeddedExtensionBuilder builder() {
+        return new EmbeddedExtensionBuilderImpl();
+    }
+
+    /**
+     * Sets the unique ID of the extension
+     * <p>
+     * Required value.
+     *
+     * @param id The unique ID of an extension
+     * @return the builder itself
+     */
+    @NotNull EmbeddedExtensionBuilder withId(@NotNull String id);
+
+    /**
+     * Sets the human readable name of the extension.
+     * <p>
+     * Required value.
+     *
+     * @param name The name of the extension
+     * @return the builder itself
+     */
+    @NotNull EmbeddedExtensionBuilder withName(@NotNull String name);
+
+    /**
+     * Sets the version of the extension.
+     * <p>
+     * Required value.
+     *
+     * @param version The version of the extension
+     * @return the builder itself
+     */
+    @NotNull EmbeddedExtensionBuilder withVersion(@NotNull String version);
+
+    /**
+     * Sets the author of the extension.
+     * <p>
+     * Optional value, no default.
+     *
+     * @param author The author of the extension
+     * @return the builder itself
+     */
+    @NotNull EmbeddedExtensionBuilder withAuthor(@Nullable String author);
+
+    /**
+     * Sets the extension's priority, the extension with a higher priority is used first
+     * <p>
+     * Optional value, defaulting to '0'
+     *
+     * @param priority The priority of the extension
+     * @return the builder itself
+     */
+    @NotNull EmbeddedExtensionBuilder withPriority(int priority);
+
+    /**
+     * Sets the extension's start priority, the extension with a higher priority starts first
+     * <p>
+     * Optional value, defaulting to '1000'
+     *
+     * @param startPriority The start priority of the extension
+     * @return the builder itself
+     */
+    @NotNull EmbeddedExtensionBuilder withStartPriority(int startPriority);
+
+    /**
+     * Sets the main class of the extension
+     * <p>
+     * Required value.
+     *
+     * @param extensionMain The main class of the extension
+     * @return the builder itself
+     */
+    @NotNull EmbeddedExtensionBuilder withExtensionMain(@NotNull ExtensionMain extensionMain);
+
+    /**
+     * Creates an instance of an EmbeddedExtension.
+     * <p>
+     * required values are:
+     * <ul>
+     * <li>id</li>
+     * <li>name</li>
+     * <li>version</li>
+     * <li>extensionMain</li>
+     * </ul>
+     *
+     * @return an instance of an {@link EmbeddedExtension}
+     * @throws NullPointerException if any required parameter is missing.
```

Comment:
imho should throw an IllegalStateException if the build is called in an invalid state with missing parameters